### DISCUSSION TITLE
NEMSIO vs. NETCDF

### DISCUSSION THREAD

**Discussion by MarcusGDaniels** [NOAA-EMC/global-workflow#1200] [Q&A]
Hi,

I'm trying to set up initial conditions using GFSv.16 and I find that one of the scripts ush/global_chgres_driver.sh wants to load nemsio files instead of netcdf as is provided here https://nomads.ncep.noaa.gov/pub/data/nccf/com/gfs/prod/gfs.20221221/00/atmos/ (for example, gfs.t00z.sfcanl.nc).   

I looked at both v16.2.2 and v16.3.4 and they both have this way of loading.  

Because I don't have access to HPSS, the instructions on the wiki won't work for me.

Thanks for any suggestions!

Marcus

**Comment by GeorgeGayno-NOAA** [comment]
The `global_chgres_driver.sh` script and its program were deprecated and removed from the official UFS_UTILS fork.

**Comment by GeorgeGayno-NOAA** [comment]
Do you have access to any NOAA RDHPCS machines?

**Comment by MarcusGDaniels** [comment]
Hi George,

No, I am standing up a build of GFS on another supercomputer.

Marcus
________________________________

**Reply by GeorgeGayno-NOAA** [reply]
Pull the NetCDF data you want to use to your machine. Then, clone and compile the latest version of UFS_UTILS 'develop'. I can help you run the 'gdas_init' utility without HPSS access.

**Reply by MarcusGDaniels** [reply]
Is there also a procedure for setting up ensembles?   I do have deterministic runs working from the S3 restart files.

**Reply by MarcusGDaniels** [reply]
Is there a listing of what the extract directory should look like after the HPSS extracting?   I can't really deduce it from the enkfgdas get_v16.data.sh logic:

```
 group=$bundle

  directory=/NCEPPROD/hpssprod/runhistory/5year/rh${yy_m6}/${yy_m6}${mm_m6}/${yy_m6}${mm_m6}${dd_m6}
  file=com_gfs_${version_enkf}_enkfgdas.${yy_m6}${mm_m6}${dd_m6}_${hh_m6}.enkfgdas_${group}.tar

  rm -f ./list*.${group}
  htar -tvf  $directory/$file > ./list1.${group}
  grep "006.nc" ./list1.${group} > ./list2.${group}
  while read -r line
  do 
    echo ${line##*' '} >> ./list3.${group}
  done < "./list2.${group}"
  htar -xvf $directory/$file  -L ./list3.${group}
  rc=$?
  [ $rc != 0 ] && exit $rc
  rm -f ./list*.${group}
```

**Reply by GeorgeGayno-NOAA** [reply]
After pulling a GFS bundle (CDUMP=gfs), the directory structure will be:

```
./gfs.20230310/00/atmos:
total 14066860
-rw-r--r-- 1 George.Gayno stmp 14052697010 Mar 10 03:23 gfs.t00z.atmanl.nc
-rw-r--r-- 1 George.Gayno stmp   351701675 Mar 10 03:23 gfs.t00z.sfcanl.nc
```

**Reply by MarcusGDaniels** [reply]
Aren't some sort of perturbed replicates needed for the ensembles?

**Reply by GeorgeGayno-NOAA** [reply]
After pulling the ENKF bundles (CDUMP=gdas), the directory structure is (this is member 1 of 80):

```
./enkfgdas.20230309/18/atmos/mem001:
total 2129444
-rw-r--r-- 1 George.Gayno stmp 1893668425 Mar 10 00:56 gdas.t18z.atmf006.nc
-rw-r--r-- 1 George.Gayno stmp  286820892 Mar 10 00:56 gdas.t18z.sfcf006.nc
```

**Reply by MarcusGDaniels** [reply]
ok, so for example, this is a sufficient substitute for what is extracted from HPSS?

https://nomads.ncep.noaa.gov/pub/data/nccf/com/gfs/prod/enkfgdas.20230320/06/atmos/mem001/

**Reply by GeorgeGayno-NOAA** [reply]
I only see the surface files on the public server.

**Reply by MarcusGDaniels** [reply]
Is there somewhere to get them?

**Reply by MarcusGDaniels** [reply]
This page says the atmf should be there (relevant part copied as image).   https://www.nco.ncep.noaa.gov/pmb/products/gfs/

Also I looked here, but again no atmf files.

https://noaa-gfs-bdp-pds.s3.amazonaws.com/index.html#enkfgdas.20230319/06/atmos/mem001/

<img width="982" alt="Screenshot 2023-03-20 at 3 44 45 PM" src="https://user-images.githubusercontent.com/350911/226482209-d9c3bf8d-dcea-4133-9374-859802eb7fe1.png">

**Reply by MarcusGDaniels** [reply]
Alternatively, is there a documented way to generate these mem*/gdas.t18z.atmf006.nc files?

**Reply by GeorgeGayno-NOAA** [reply]
I don't know of a way to generate those files on your own.

**Reply by MarcusGDaniels** [reply]
Any suggestions on who to ask?  It seems like a bug these were dropped.

**Reply by GeorgeGayno-NOAA** [reply]
I don't think it is a bug. It is a lack of disk space. There is just no room for the 80 ENKF atmospheric files.

**Reply by MarcusGDaniels** [reply]
It isn't possible to independently reproduce the ensemble predictions.  That can't be a good thing from a scientific perspective?  Also, why share half of them?   It seems to be a waste of that disk space too.   At one point according to the screenshot above, they were being shared.  It seems as though they were pulled back.

**Reply by MarcusGDaniels** [reply]
> Pull the NetCDF data you want to use to your machine. Then, clone and compile the latest version of UFS_UTILS 'develop'. I can help you run the 'gdas_init' utility without HPSS access.

Hi, I have both the atmospheric and surface netcdf ensemble sets now.   What is the next step?

**Comment by aerorahul** [comment]
@MarcusGDaniels 
If you have need for a single case of initial conditions, we can provide.
However, please note that this data is not publicly available due to its size.
https://nomads.ncep.noaa.gov/pub/data/nccf/com/gfs/prod/ is the holding tank for a few days of current data.
The ensemble data would be under `enkfgdas`, but as you note is not complete for your needs.
There must have been a Public Notification Statement issued when it was decided to cut down the data volume.  A PNS typically has a window to gather public comment.

**Reply by MarcusGDaniels** [reply]
A single case of initial conditions would certainly be of interest.   If funding is needed for more storage, I can investigate getting it.

**Comment by MarcusGDaniels** [comment]
It seems like there must be code somewhere for generating the 80 variants?

**Reply by aerorahul** [reply]
The code (chgres_cube) to downscale the 80 members exists in:
https://github.com/ufs-community/UFS_UTILS
The code operates on 80 members at any given resolution.
These 80 members (typically) come from the operational cycled data assimilation system.

The issue here is access to the raw data from the cycled data assimilation system that is not publicly accessible.

We have 80 members at C48, if that is acceptable to your needs that I can easily host temporarily on an FTP site.  Would that be sufficient for your needs?

**Reply by MarcusGDaniels** [reply]
We'd like to do production uncertainty characterization at C768.  C48 would enable us to rationalize the workflow, which would be valuable.   

What I mean about the 80 is if there exists code to expand a single gdas set into a set of 80?  Is it correct to say that each of the 80 members is "raw" data?  Or is each member derived by an algorithm?

**Reply by MarcusGDaniels** [reply]
Will the rocoto workflow of eobs/ediag/eomg/eupd/econ/esfc/efcs/echgres/epos do the job?  (Assuming I could get that to work.)
Or will there still be dependencies missing?

**Reply by MarcusGDaniels** [reply]
How was this process bootstrapped?   Would it converge toward the set of mem*/atmfXXX files you have at C768 starting with C48 or just copies of a sfc and atm files like at the URL below?   How much disk space is it, in fact?   Do you offer a way to buy this data as a subscription?

https://nomads.ncep.noaa.gov/pub/data/nccf/com/gfs/prod/gdas.20230314/00/atmos/

**Reply by MarcusGDaniels** [reply]
Hi again,

Could I try the 80 of C48?   

Is it the case that chgres will let me mix those with the full resolution surface data?  
If so, might it possible to make those C48 runs part of the workflow so they would be published each day?

Thanks,

Marcus

**Reply by MarcusGDaniels** [reply]
In case the issue is S3 bandwidth costs, I can provide a login to do a direct rsync.
